railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -137,7 +144,7 @@ def polymorphic?
       end
 
       def required?
-        attr_options[:required]
+        reference?
```

Comment:
Can't people still set `belongs_to_required_by_default = false`? Don't think we have deprecated that or necessarily plan to.

Reply:
@kaspth No, it can still be used https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations/builder/belongs_to.rb#L115

It only controls validations as of now. So even if `required` is true set here, validation is still controlled by the global config. 

But because of changes in this PR, the global config is not controlling `null:false`. I will make the change for it and push.